### PR TITLE
DB implementations in chain 

### PR DESCRIPTION
**Pull request by ouwenkg**
### Description of the Change

1. implement all unimplemented function use `cita_db`
2. ~~remove old `db.rs`~~
3. rename `TransactionAddress` as `TransactionIndex`

### CODE REVIEW

File: cita-chain/types/src/extras.rs [hunk 1]
```diff
@@ -176,11 +176,11 @@ fn with_index(hash: &H256, i: ExtrasIndex) -> H264 {
     result
 }
 
-impl Key<TransactionAddress> for H256 {
```
Comment (jerry-yu) [inline cita-chain/types/src/extras.rs]: 换一下名字，用Index 替代 Key  trait ？

Comment (ouwenkg) [reply cita-chain/types/src/extras.rs]: Key trait 这里应该要重新写，改名字，解决不了问题。

Comment (ouwenkg) [reply cita-chain/types/src/extras.rs]: > 换一下名字，用Index 替代 Key trait ？

修改了一下， 再重新 review 一下吧。

File: cita-chain/types/src/block.rs [hunk 2]
```diff
@@ -206,12 +206,12 @@ impl BlockBody {
...
 
-    pub fn transaction_addresses(&self, hash: H256) -> HashMap<H256, TransactionAddress> {
+    pub fn transaction_indexes(&self, hash: H256) -> HashMap<H256, TransactionIndex> {
         let tx_hashs = self.transaction_hashes();
-        // Create TransactionAddress
+        // Create TransactionIndex
         let mut transactions = HashMap::new();
         for (i, tx_hash) in tx_hashs.into_iter().enumerate() {
-            let address = TransactionAddress {
+            let address = TransactionIndex {
```
Comment (kaikai1024) [inline cita-chain/types/src/block.rs]: address -> index

File: cita-chain/core/src/libchain/chain.rs [hunk 3]
```diff
@@ -253,8 +238,15 @@ impl Config {
 }
 
 impl BloomGroupDatabase for Chain {
-    fn blooms_at(&self, _position: &BloomGroupPosition) -> Option<BloomGroup> {
-        unimplemented!()
+    fn blooms_at(&self, position: &BloomGroupPosition) -> Option<BloomGroup> {
+        let p = LogGroupPosition::from(position.clone());
```
Comment (kaikai1024) [inline cita-chain/core/src/libchain/chain.rs]: Why not use `let position = LogGroupPosition::from(position.clone());`

Comment (ouwenkg) [reply cita-chain/core/src/libchain/chain.rs]: There is a local variable already named `position`.  In my view, an abbreviated variable name is acceptable while expliciting variable type.

File: cita-chain/types/src/block.rs (hunk 2 above)
Comment (ouwenkg) [reply cita-chain/types/src/block.rs]: Done

File: cita-chain/core/src/libchain/chain.rs (hunk 3 above)
Comment (kaikai1024) [reply cita-chain/core/src/libchain/chain.rs]: > There is a local variable already named position.

yeah, i see it. that's ok for this function.

> variable bindings can be shadowed.

> an abbreviated variable name is acceptable

I don't agree. but that's ok for me. Forget it.

File: cita-auth/src/txwal.rs [hunk 4]
```diff
@@ -87,7 +87,7 @@ impl TxWal {
             .collect()
     }
 
-    pub fn get(&self, _tx_hash: &[u8]) -> Option<SignedTransaction> {
```
Comment (kaikai1024) [inline cita-auth/src/txwal.rs]: A new pr is better to fix the building.
no need to be blocked by this pr.

**Comment by kaikai1024** [comment]
This `[types] remove db.rs` commit can drop if :

> ~~remove old db.rs~~

**Comment by ouwenkg** [comment]
> This `[types] remove db.rs` commit can drop if :
> 
> > ~remove old db.rs~

Done

**Pull request merged**